source/non_matching/quadrature_generator.cc

```diff
@@ -1896,7 +2138,76 @@ namespace NonMatching
 
     reference_space_level_set->set_active_cell(cell);
     const BoundingBox<dim> unit_box = create_unit_bounding_box<dim>();
-    QuadratureGenerator<dim>::generate(*reference_space_level_set, unit_box);
+    if (reference_space_level_set->is_fe_q_iso_q1)
+      {
+        const unsigned int n_subdivisions =
+          reference_space_level_set->n_subdivisions;
+
+        const unsigned int dofs_per_line = n_subdivisions + 1;
+
+        this->clear_quadratures();
+
+        auto generate =
+          [&](const std::array<unsigned int, dim> &subcell_indices) {
+            const std::vector<unsigned int> lexicographic_mask =
+              internal::DiscreteQuadratureGeneratorImplementation::
+                setup_lexicographic_mask<dim>(subcell_indices, dofs_per_line);
+
+            reference_space_level_set->set_subcell_mask(lexicographic_mask);
+
+            const auto subcell_box =
+              internal::DiscreteQuadratureGeneratorImplementation::
+                create_subcell_box<dim>(unit_box,
+                                        subcell_indices,
+                                        n_subdivisions);
+
+            reference_space_level_set->set_subcell_box(subcell_box);
+
+            QuadratureGenerator<dim>::generate_append(
+              *reference_space_level_set, subcell_box);
+          };
+
+        if (dim == 1)
```

Comment:
It would be nice if we could clean this up in some way, but I'm not really sure how. Maybe pull it into a function?
```c++
void traverse_all_subcell_boxes(
const unsigned int n_subdivisions,
std::function<...> &generate)
```